router/handler_ctrl/route.go

```diff
@@ -83,22 +85,36 @@ func (rh *routeHandler) HandleReceive(msg *channel.Message, ch channel.Channel)
 		if route.Egress != nil {
 			if rh.forwarder.HasDestination(xgress.Address(route.Egress.Address)) {
 				log.Warnf("destination exists for [%s]", route.Egress.Address)
-				rh.success(msg, int(route.Attempt), route, nil, log)
+				rh.completeRoute(msg, int(route.Attempt), route, nil, log)
 				return
 			} else {
 				rh.connectEgress(msg, int(route.Attempt), ch, route, ctx)
 				return
 			}
 		} else {
-			rh.success(msg, int(route.Attempt), route, nil, log)
+			rh.completeRoute(msg, int(route.Attempt), route, nil, log)
 		}
 	} else {
 		pfxlog.ContextLogger(ch.Label()).WithError(err).Error("error unmarshaling")
 	}
 }
 
-func (rh *routeHandler) success(msg *channel.Message, attempt int, route *ctrl_pb.Route, peerData xt.PeerData, log *logrus.Entry) {
-	rh.forwarder.Route(route)
+func (rh *routeHandler) completeRoute(msg *channel.Message, attempt int, route *ctrl_pb.Route, peerData xt.PeerData, log *logrus.Entry) {
+	if err := rh.forwarder.Route(route); err != nil {
+		var ide forwarder.InvalidDestinationError
+		if errors.As(err, &ide) {
+			linkId := string(ide)
+			if strings.HasSuffix(linkId, "L") {
```

Comment:
strings.HasSuffix(linkId, "L") seems like it should be a function imo /shrug

Reply:
Fixed, see above.